### DISCUSSION TITLE
Why is it necessary to not use the odometry thread when using vision?

### DISCUSSION THREAD

**Discussion by mtappen** [BroncBotz3481/YAGSL-Example#274] [Q&A] [answered]
I noticed that YAGSL-Example stops the odometry thread when using vision.  Why is that necessary?  There is a lock in the odometry thread.  Couldn't the vision update code also use the same lock so that odometry is only updated from one source at a time?

**Comment by clrozeboom** [comment]
I've wondered about this myself. In fact the vision update functions do use the lock. In my own code I don't stop the thread and it seems to work fine.

**Reply by mtappen** [reply]
Thanks, I didn't notice that the vision measurement functions did use the lock.

**Comment by thenetworkgrinch** [comment, marked as answer]
It should lessen the interrupts ideally, its not necessary but is an optimization
